iroh-bytes/src/store/flat.rs

```diff
@@ -1430,38 +1544,30 @@ impl Store {
             }
         }
         for hash in complete.keys() {
-            tracing::debug!("complete {}", hash);
+            tracing::info!("complete {}", hash);
             partial.remove(hash);
         }
         for hash in partial.keys() {
-            tracing::debug!("partial {}", hash);
+            tracing::info!("partial {}", hash);
         }
-        let tags_path = meta_path.join("tags.meta");
-        let mut tags = BTreeMap::new();
-        if tags_path.exists() {
-            let data = std::fs::read(tags_path)?;
-            tags = postcard::from_bytes(&data)?;
-            tracing::debug!("loaded tags. {} entries", tags.len());
-        };
-        Ok(Self(Arc::new(Inner {
-            state: RwLock::new(State {
-                complete,
-                partial,
-                outboard,
-                data: Default::default(),
-                live: Default::default(),
-                temp: Default::default(),
-            }),
-            tags: RwLock::new(tags),
-            options: Options {
-                complete_path,
-                partial_path,
-                meta_path,
-                move_threshold: 1024 * 128,
-                inline_threshold: 1024 * 16,
-            },
-            complete_io_mutex: Mutex::new(()),
-        })))
+        let txn = self.0.db.begin_write()?;
+        {
+            let mut complete_table = txn.open_table(COMPLETE_TABLE)?;
+            let mut partial_table = txn.open_table(PARTIAL_TABLE)?;
+            let mut tags_table = txn.open_table(TAGS_TABLE)?;
+            for (hash, entry) in complete {
+                complete_table.insert(hash, entry)?;
+            }
+            for (hash, entry) in partial {
+                partial_table.insert(hash, entry)?;
+            }
+            for (tag, target) in tags {
```

Comment:
shouldn't this delete the tags and external path files on successs?

Reply:
yeah.